dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/TemplatePage.java

```diff
@@ -113,45 +113,49 @@ public void setOwnerId(Long ownerId) {
 
     public String init() {
 
-        dataverse = dataverseService.find(ownerId);
-        if (dataverse == null) {
-            return permissionsWrapper.notFound();
-        }
-        if (!permissionsWrapper.canIssueCommand(dataverse, UpdateDataverseCommand.class)) {
-            return permissionsWrapper.notAuthorized();
-        }
-        if (templateId != null) { // edit existing template
+        if (templateId != null) {
             editMode = TemplatePage.EditMode.METADATA;
             template = templateService.find(templateId);
-            template.setDataverse(dataverse);
+
+            dataverse = template.getDataverse();
+
+            String validationError = validatePermissionsForTemplateEdit(dataverse, templateId);
+
+            if (!validationError.isEmpty()) {
+                return validationError;
+            }
```

Comment:
I'm not sure If it will work as we could expect based on `NavigationWrapper.notAuthorized()` can return empty string